crates/ruff_linter/src/rules/pycodestyle/rules/blank_lines.rs

```diff
@@ -602,40 +611,85 @@ impl Status {
 }
 
 /// Contains variables used for the linting of blank lines.
-#[derive(Debug, Default)]
-pub(crate) struct BlankLinesChecker {
-    follows: Follows,
-    fn_status: Status,
-    class_status: Status,
-    /// First line that is not a comment.
-    is_not_first_logical_line: bool,
-    /// Used for the fix in case a comment separates two non-comment logical lines to make the comment "stick"
-    /// to the second line instead of the first.
-    last_non_comment_line_end: TextSize,
-    previous_unindented_line_kind: Option<LogicalLineKind>,
+#[derive(Debug)]
+pub(crate) struct BlankLinesChecker<'a> {
+    stylist: &'a Stylist<'a>,
+    locator: &'a Locator<'a>,
+    indent_width: IndentWidth,
+    isort_enabled: bool,
 }
 
-impl BlankLinesChecker {
+impl<'a> BlankLinesChecker<'a> {
+    pub(crate) fn new(
+        locator: &'a Locator<'a>,
+        stylist: &'a Stylist<'a>,
+        settings: &crate::settings::LinterSettings,
+    ) -> BlankLinesChecker<'a> {
+        BlankLinesChecker {
+            stylist,
+            locator,
+            indent_width: settings.tab_size,
+            isort_enabled: settings.rules.enabled(Rule::UnsortedImports),
+        }
+    }
+
     /// E301, E302, E303, E304, E305, E306
-    pub(crate) fn check_lines(
-        &mut self,
-        tokens: &[LexResult],
-        locator: &Locator,
-        stylist: &Stylist,
-        indent_width: IndentWidth,
-        diagnostics: &mut Vec<Diagnostic>,
-    ) {
+    pub(crate) fn check_lines(&self, tokens: &[LexResult], diagnostics: &mut Vec<Diagnostic>) {
         let mut prev_indent_length: Option<usize> = None;
-        let line_preprocessor = LinePreprocessor::new(tokens, locator, indent_width);
+        let mut state = BlankLinesState::default();
+        let line_preprocessor = LinePreprocessor::new(tokens, self.locator, self.indent_width);
 
         for logical_line in line_preprocessor {
-            self.check_line(
-                &logical_line,
-                prev_indent_length,
-                locator,
-                stylist,
-                diagnostics,
-            );
+            state.class_status.update(&logical_line);
```

Comment:
The state update logic used to be inside of `check_line`. Moving the state handling out has the benefit that we can skip calling `check_line` and it also separates the state update logic from the actual checking logic which I find easier to understand.